common/pkg/cgroups/cgroups_linux.go

```diff
@@ -485,50 +291,25 @@ func (c *CgroupControl) DeleteByPath(path string) error {
 
 // Update updates the cgroups.
 func (c *CgroupControl) Update(resources *cgroups.Resources) error {
-	for _, h := range handlers {
-		if err := h.Apply(c, resources); err != nil {
-			return err
-		}
+	man, err := fs2.NewManager(c.config, filepath.Join(cgroupRoot, c.config.Path))
+	if err != nil {
+		return err
 	}
-	return nil
+	return man.Set(resources)
 }
 
 // AddPid moves the specified pid to the cgroup.
 func (c *CgroupControl) AddPid(pid int) error {
-	pidString := []byte(fmt.Sprintf("%d\n", pid))
-
-	if c.cgroup2 {
-		path := filepath.Join(cgroupRoot, c.config.Path)
-		return fs2.CreateCgroupPath(path, c.config)
-	}
-
-	names := slices.Collect(maps.Keys(handlers))
-
-	for _, c := range c.additionalControllers {
-		if !c.symlink {
-			names = append(names, c.name)
-		}
-	}
-
-	for _, n := range names {
-		// If we aren't using cgroup2, we won't write correctly to unified hierarchy
-		if !c.cgroup2 && n == "unified" {
-			continue
-		}
-		p := filepath.Join(c.getCgroupv1Path(n), "tasks")
-		if err := os.WriteFile(p, pidString, 0o644); err != nil {
-			return fmt.Errorf("write %s: %w", p, err)
-		}
-	}
-	return nil
+	path := filepath.Join(cgroupRoot, c.config.Path)
+	return fs2.CreateCgroupPath(path, c.config)
```

Comment:
@giuseppe could you PTAL at this diff please? You mentioned AddPid is required, but if it's only cgroup2, then I think this could be simplified as the current diff, but in that case the `pid` isn't even being used. I'll defer to you.